### DISCUSSION TITLE
Custom flow field rendering overlay on osgEarth

### DISCUSSION THREAD

**Discussion by FuXiii** [pelicanmapping/osgearth#2572] [Support Q&A] [answered]
I had a custom flow field rendering use `instanced` draw in `OSG`, it looks like :
  
https://github.com/gwaldron/osgearth/assets/40132375/449db7a3-fefd-4b69-92ac-14b847607aa7

Just `instanced` massive `points` and transform them on earth surface.

Then I want combine them with `osgEearth` , and the `scene tree`  may looks like :

```text
root
 |_ earth
 |_ flowField
```

https://github.com/gwaldron/osgearth/assets/40132375/22bdad94-aad9-404b-9394-7a313d96c6b6

Looks like `earth` overlap the `flow field` (`flow field` draw first, `earth` draw second), I try to set `RenderBinDetials` of `flow field` :

```cpp
flowFieldNode->getOrCreateStateSet()->setRenderBinDetials(1000, "RenderBin");
```
But it didn't solve the problem.

How to draw my custom things after `earth` had finished drawn (Let `flow field` overlay on `earth`).

Thanks~

**Comment by FuXiii** [comment]
I had recorded a new video to describe my issue:

This time I just draw the flow field with quarter of earth.

And the flow field instanced draw method are very simple. It just draw many point as usual. And `StateSet` looks like:

![Snipaste_2024-07-03_15-33-43](https://github.com/gwaldron/osgearth/assets/40132375/8a8258ed-c2c2-4d50-bac2-17c6a4c118a5)

## Result
https://github.com/gwaldron/osgearth/assets/40132375/2d5a0946-708b-4d11-a80a-5c6ae84b162a

https://github.com/gwaldron/osgearth/assets/40132375/427e340f-b803-4ec3-92f1-20ac1e27ce20

Looks like `osgEarth` will change earth `pixel depth` dynamically? Is there some method that I can draw common things overlay(coverd) on earth?

**Comment by gwaldron** [comment, marked as answer]
It seems like you are having depth-testing issues. Things to try:
1. Draw your points farther away from the earth's surface.
2. Try the `LogarithmicDepthBuffer` for higher depth precision.
3. Draw your points with depth testing disabled, and then use your shader to discard points that are on the opposite side of the earth from your camera.

Good luck!

**Reply by FuXiii** [reply]
Yes, The third proposal is a good suggestion for me. Thank you very much!
